app/models/concerns/attachmentable.rb

```diff
@@ -75,9 +75,16 @@ def appropriate_extension(attachment)
   end
 
   def calculated_content_type(attachment)
-    Paperclip.run('file', '-b --mime :file', file: attachment.queued_for_write[:original].path).split(/[:;\s]+/).first.chomp
-  rescue Terrapin::CommandLineError
-    ''
+    begin
+      Timeout.timeout(10) do
```

Comment:
I wish it was this simple. `Timeout` creates 2 new threads, one with the timer and one with the code block, then kills the latter after the timer reaches the timeout. With Sidekiq this means potentially a lot more threads will be spawned than your machine is ready for, if every Sidekiq thread will spawn 2 more threads. In our `Request` class, we avoid this by using a low-level timeout features of `Resolv` and `Socket`. It seems like there is no timeout feature for the `file` command specifically though.

Reply:
I wish it were this simple, too :) Thanks for clarifying. I will close this PR and will keep trying to find more information regarding #34716.